Esmorga-Backend/esmorga-backend
# [MOB-40] Change config for Jest tests

## Pull Request Checklist

- [x] **MOB US' Link**
- [x] **[Link PR to US](https://mobilemakers.atlassian.net/browse/MOB-40)**

## Testing Guide
```
npm i
```
New scripts added:
- `"test:unit": "jest --config ./test/unit/jest-unit.config.json --detectOpenHandles",` &rarr; will trigger **only** unit test
- `"test:cov:unit": "jest --config ./test/unit/jest-unit.config.json --coverage --detectOpenHandles",` &rarr; will trigger **only** unit test and generate a report 
- `"test:component": "jest --config ./test/component/jest-component.config.json --detectOpenHandles --forceExit",` &rarr; will trigger *only* component test and generate a report 
- `"test:cov:component": "jest --config ./test/component/jest-component.config.json --coverage --detectOpenHandles  --forceExit",` &rarr; 
- `"test:merge-coverage": "istanbul-merge --out coverage/merged-output/merged-coverage.json coverage/unit/coverage-final.json coverage/component/coverage-final.json",` &rarr; combine unit and component tests results in one json file
- `"test:report": "nyc report -t coverage/merged-output --report-dir coverage --reporter=html --reporter=cobertura --reporter=lcov",` &rarr; generate the report with the mixed data
- `"test": "npm run test:cov:unit && npm run test:cov:component && npm run test:merge-coverage && npm run test:report",` &rarr; execute previous command mentioned to check both tests types and generate the report with the coverage data
